web/src/components/modals/standardize-metadata.tsx

```diff
@@ -0,0 +1,424 @@
+        [key]: value === null ? key : value,
+      };
+      setWhereDuplicates(checkForDuplicates(newValues));
+      return newValues;
+    });
+  };
+
+  const checkForDuplicates = (values: SelectedValues): number[] | null => {
+    const valueArray = Object.values(values);
+    const duplicates: Record<string, number[]> = {};
+    const result: number[] = [];
+
+    for (let i = 0; i < valueArray.length; i++) {
+      const value = valueArray[i];
+      if (value in duplicates) {
+        if (duplicates[value].length === 1) {
+          result.push(duplicates[value][0]);
+        }
+        result.push(i);
+      } else {
+        duplicates[value] = [i];
+      }
+    }
+
+    return result.length > 0 ? result : null;
+  };
+
+  const getDefaultSelections = (standardizedData: StandardizedData): SelectedValues => {
+    const defaultSelections: SelectedValues = {};
+    Object.keys(standardizedData).forEach((key) => {
+      defaultSelections[key] = key;
+    });
+    return defaultSelections;
+  };
+
+  const updateTabDataRaw = (tabDataRaw: TabData, selectedValues: SelectedValues): TabData => {
+    if (tabDataRaw.length === 0) return tabDataRaw;
+
+    const updatedTabDataRaw: TabData = [tabDataRaw[0].slice(), ...tabDataRaw.slice(1)];
+
+    Object.entries(selectedValues).forEach(([key, value]) => {
+      let columnIndex = updatedTabDataRaw[0].indexOf(key);
+      if (columnIndex === -1) {
+        const originalValue = getOriginalColValues(key);
+        if (originalValue) {
+          columnIndex = updatedTabDataRaw[0].indexOf(originalValue);
+        }
+      }
+      if (columnIndex !== -1) {
+        updatedTabDataRaw[0][columnIndex] = value;
+      }
+    });
+
+    return updatedTabDataRaw;
+  };
+
+  const handleSubmit = async (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    setResetStandardizedData(true);
+    await standardize();
+  };
+
+  const prepareHandsontableData = useCallback(
+    (key: string) => {
+      const selectedValue = selectedValues[key] || '';
+      const originalValue = getOriginalColValues(key);
+      let topValues: string[][] = [];
+      if (originalValue) {
+        topValues =
+          tabData[key]?.slice(0, 6).map((item) => [item]) ||
+          tabData[originalValue]?.slice(0, 6).map((item) => [item]) ||
+          [];
+      }
+      const emptyRows = Array(Math.max(0, 6 - topValues.length)).fill(['']);
+
+      return [[selectedValue], ...topValues, ...emptyRows];
+    },
+    [selectedValues, tabData],
+  );
+
+  useEffect(() => {
+    if (standardizedData) {
+      const defaultSelections = getDefaultSelections(standardizedData);
+      setWhereDuplicates(checkForDuplicates(defaultSelections));
+      setSelectedValues(defaultSelections);
+    }
+  }, [standardizedData]);
+
+  useEffect(() => {
+    if (!resetStandardizedData) {
+      setSelectedOption(null);
+      setSelectedValues({});
+      setWhereDuplicates(null);
+    }
+  }, [resetStandardizedData]);
```

Comment:
Possible to get rid of this? I think I see whats going on, and I think that we can meet to discuss how to circumvent the issue... if you would like. Its probably fine for now, but I try to remove all `useEffects` when I see them

Reply:
Not really sure.. Can we push this to the next update?